lib/ansible/module_utils/basic.py

```diff
@@ -265,8 +265,8 @@ def _get_available_hash_algorithms():
 
 # Used for parsing symbolic file perms
 MODE_OPERATOR_RE = re.compile(r'[+=-]')
-USERS_RE = re.compile(r'[^ugo]')
-PERMS_RE = re.compile(r'[^rwxXstugo]')
+USERS_RE = re.compile(r'[ugo]+')
```

Comment:
I note that you have "+" in the USERS_RE, but "*" in the PERMS_RE.  Note that the users portion of a symbolic permission can be empty, for example: "=rwx".

Reply:
@linsomniac Yes, it can be passed in as empty. Right before the regex is used we replace an empty users string https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1061-L1062, so it should be 1 or more. Thanks for the review.